src/java/org/apache/cassandra/index/sai/utils/IndexTermType.java

```diff
@@ -275,6 +275,14 @@ public boolean isValid(ByteBuffer term)
         }
     }
 
+    /**
+     * @return {@code true} if the empty values of the given type should be excluded from indexing
+     */
+    public boolean skipsEmptyValue()
+    {
+        return !indexType.allowsEmpty() || !isLiteral();
```

Comment:
The alternative here is to use `isEmptyValueMeaningless()` instead of `!isLiteral()`, which would mean empty values for boolean and a few other things would no longer be indexed. This makes sense if you don't believe querying on empty values for non-string types is useful, but it obviously diverges from what non-index filtering queries seem to do right now.